### PR TITLE
Update docs based on crate restructuring

### PR DESCRIPTION
**Pull request by seanmlyons22**
* Install target folder is now probe-rs-tools
* Fixed dead link to default cargo-embed toml
* Associated PR: https://github.com/probe-rs/probe-rs/pull/2431

### CODE REVIEW

File: src/docs/tools/cargo-embed.md [hunk 1]
```diff
@@ -60,7 +60,7 @@ enabled = true
 ```
 
 All available options can be found in the
-[default.toml](https://github.com/probe-rs/probe-rs/blob/master/probe-rs/src/bin/probe-rs/cmd/cargo_embed/config/default.toml).
+[default.toml](https://github.com/probe-rs/probe-rs/blob/master/probe-rs-tools/src/bin/probe-rs/cmd/cargo_embed/config/default.toml.toml).
```
Comment (bugadani) [inline src/docs/tools/cargo-embed.md:63]:
```suggestion
[default.toml](https://github.com/probe-rs/probe-rs/blob/master/probe-rs-tools/src/bin/probe-rs/cmd/cargo_embed/config/default.toml).
```

**Review by bugadani — DISMISSED** [review]

File: src/docs/library/crosscompiling.md [hunk 2]
```diff
@@ -52,17 +52,17 @@ vim Cross.toml
...
+image = "crossimage"
+```
 
 ```sh
 # Build and tag the container image, specifying the name as defined in the `Cross.toml`.
 docker build -t crossimage crossimage/
 
 # Run cross to compile, cross arguments are the same as the `cargo` ones
-cross build --release --target=armv7-unknown-linux-gnueabihf --features cli
+cross build --path probe-rs-tools --release --target=armv7-unknown-linux-gnueabihf
```
Comment (bugadani) [inline src/docs/library/crosscompiling.md:65]: @Dirbaio I believe you edited this last time. Can you tell me why the path wasn't specified in the old revision (consequently, can you tell me if the new version is correct?)?

**Comment by bugadani** [comment]
@seanmlyons22 sorry for the delay. I've taken the liberty to update the PR with some more changes, I hope you don't mind.

@Yatekii I think this should be suffcient for the website part of the new release.

File: src/docs/library/crosscompiling.md (hunk 2 above)
Comment (Yatekii) [reply src/docs/library/crosscompiling.md:65]: I think we had a default that pointed to probe-rs

File: src/docs/tools/cargo-flash.md [hunk 3]
```diff
@@ -18,10 +18,10 @@ You can use **cargo-flash** just like 'cargo run', but instead of running on the
 
 Here are the basics
 
-```sh
-# Installing cargo flash is simple:
-cargo install probe-rs --features cli
+**cargo-embed** is installed as part of the **probe-rs** group of tools, see the
```
Comment (Yatekii) [inline src/docs/tools/cargo-flash.md:21]:
```suggestion
**cargo-flash** is installed as part of the **probe-rs** group of tools, see the
```

File: src/docs/library/crosscompiling.md (hunk 2 above)
Comment (bugadani) [reply src/docs/library/crosscompiling.md:65]: I don't see how that's specified :/

Comment (Yatekii) [reply src/docs/library/crosscompiling.md:65]: I think we are good anyways :)

**Review by Yatekii — APPROVED** [review]

**Pull request merged**